packages/lib-classifier/src/components/Classifier/components/Feedback/components/Graph2dRangeFeedback.js

```diff
@@ -72,46 +72,39 @@ class Graph2dRangeFeedback extends Component {
       .insert('g', '.brush')
       .attr('class', 'brush')
       .attr('id', (brush) => (`brush-${brush.id}`))
-      .each(function applyBrushLogic (feedbackBrush) { // Don't use ()=>{}
-        feedbackBrush.brush(d3.select(this)) // Apply the brush logic to the <g.brush> element (i.e. 'this')
-      })
-
-    // Modify brush fill color...
-    brushSelection
-      .each(function fill (feedbackBrush) { // Don't use ()=>{}
-        d3.select(this)
-          .attr('class', 'brush')
-          .selectAll('.selection')
-          .style('fill', () => {
-            if (feedbackBrush.success === true) {
-              return 'green'
-            } else if (feedbackBrush.success === false) {
-              return 'red'
-            } else {
-              return 'white'
-            }
-          })
+      .each(function applyBrushLogic(feedbackBrush) { // Don't use ()=>{}
+        feedbackBrush.brush(select(this)) // Apply the brush logic to the <g.brush> element (i.e. 'this')
       })
 
     // Reposition/re-draw brushes
     feedbackBrushes.forEach((feedbackBrush) => {
       const d3brush = d3annotationsLayer.select(`#brush-${feedbackBrush.id}`)
+      d3brush
+        .attr('class', 'brush')
+        .selectAll('.selection')
+        .style('fill', () => {
+          if (feedbackBrush.success === true) {
+            return 'green'
+          } else if (feedbackBrush.success === false) {
+            return 'red'
+          } else {
+            return 'white'
+          }
+        })
```

Comment:
The biggest change is that the "Set Brush Colour" logic is now moved from `brushSelection` to `feedbackBrushes`, got it. I'm still trying to grok the differences between these two. From my first read, I see that...

feedbackBrushes = array of brushes (the 2D value ranges) consisting of user-made annotations (annotationBrushes) plus the feedback rules

brushSelection = the logically-OR-joined version of existing brushes (as in DOM elements) on the chart plus feedbackBrushes

What the?

I mean, the new code makes sense now, I'm just trying to figure out how the old code worked previously. 🤔

Reply:
Can I delete `brushSelection` now that it isn't used? I'm not 100% sure what it does, so I'm hesitant to remove it in case it does something vital elsewhere.

Reply:
Answering my own question: I removed `brushSelection` from the code and feedback disappeared.

Reply:
Oh man, yeah, D3.js can be gobbledygook if you're just looking at it like a normal human being. This is what's happening...

_(Technical Trivia ahead! Ignore if you can't be bothered with D3.js!)_

```
// Join the D3 brush objects with our internal annotationBrushes array
const brushSelection = d3annotationsLayer
  .selectAll('.brush')
  .data(feedbackBrushes, (d) => d.id)
```

This bit here says,
- OK, in the Annotations Layer (the actual SVG/DOM element), find all elements with class='brushes'.
- We're going to logically bind/join all those SVG elements to the data values of Feedback brushes.
- ...there's be _no visible changes_ (that happens later), we're just saying, "hey, these SVG elements are joined to these data values..."
- (And yes, confusingly, there can be cases where an existing SVG element has lost the data value it was joined to, or a data value doesn't have an existing SVG element to join to. That's where .exit() and .enter() play their part.)

```
// Set up new brushes
brushSelection.enter()
  .insert('g', '.brush')
  .doBlahBlahBlah()
```

This bit here says,
- For every NEW data value (i.e. a data value that hasn't been paired with an SVG element)...
- ...insert a new `<g class='brush'>` SVG element
- And then for each of those _new_ SVG elements, do blah blah blah on them.
- (Any SVG Element that has already been paired with an existing data value doesn't get blah blah blah performed on them.)

Removing all the `brushSelection` would result in no data points being rendered, as the brushes (data values) wouldn't be data-joined to the, erm, other brushes (SVG elements controlled by D3.js)